docs/okeps/okep-5259-no-overlay.md

```diff
@@ -0,0 +1,1104 @@
+SNAT behavior, we introduce a new `natOutbound` field to the
+ClusterUserDefinedNetwork CRD. When enabled, pod egress traffic will be SNATed
+to the node IP before leaving the node.
+
+The `natOutbound` field is intended for deployments where a user wants a
+no-overlay network but has not advertised the pod network to external BGP
+routers. When enabled, pod-to-external traffic is SNATed and forwarded by a
+default gateway, while pod-to-remote-pod traffic is not SNATed.
+
+##### Shared Gateway Mode
+
+If `natOutbound` is enabled for a network, in the gateway router, the default
+SNAT rule will be kept for pod-to-external traffic. But pod-to-remote-pod
+traffic remains unaffected by SNAT.
+
+In the future, if any non-BGP routes in the gateway router have prefixes that
+overlap with those learnt via BGP, we can add these prefixes to another address
+set and apply SNAT to traffic routed through them.
+
+In the following example, `10.128.0.0/24` is the pod subnet in the local node,
+`10.128.0.0/16` is the CIDR of the pod network. `natOutbound` is enabled.
+
+```text
+# Address_Set for the no-overlay pod network CIDR
+_uuid               : a718588e-0a9b-480e-9adb-2738e524b82d
+addresses           : ["10.128.0.0/16"]
+external_ids        : {ip-family=v4}
+name                : a15397791517117706455
+
+# per-pod SNAT rule in the Gateway Router (exclude pod-network destinations)
+_uuid               : cd9264de-f2c6-4528-8f53-c2ccb5b23c8d
+allowed_ext_ips     : []
+exempted_ext_ips    : ["a15397791517117706455"] # Exclude traffic to the pod Network
+external_ids        : {}
+external_ip         : "172.18.0.2"         # Node/GR external IP used for SNAT
+external_mac        : []
+external_port_range : "32768-60999"
+gateway_port        : []
+logical_ip          : "10.128.0.7"         # The local pod IP
+logical_port        : []
+match               : ""
+options             : {stateless="false"}
+priority            : 0
+type                : snat
+```
+
+##### Local Gateway Mode
+
+In the following example, `10.128.0.0/24` is the pod subnet in the local node,
+`10.128.0.0/16` is the CIDR of the pod network. `natOutbound` is enabled in for
+the network.
+
+```text
+table ip nat {
+    chain postrouting {
+        type nat hook postrouting priority 100; policy accept;
+        # skip SNAT for pod-to-pod traffic
+        ip saddr 10.128.0.0/24 ip daddr 10.128.0.0/16 return
+        ip saddr 10.128.0.0/24 masquerade
+    }
+}
+```
+
+#### UDN Traffic Isolation
+
+Currently, for cross-UDN pod-to-pod traffic, OVN-Kubernetes supports 2 modes:
+loose and strict. In the loose mode, cross-UDN traffic is allowed if an external
+router can route the traffic correctly. In strict mode, cross-UDN traffic is
+blocked by ACL rule defined in the node switch. In no-overlay mode, these two
+isolation modes shall still operate in the same way.
+
+### Workflow
+
+#### Enable No-overlay Mode for the Default Network
```

Comment:
So this is the complete workflow for external reflectors. Did we drop the idea of internal route reflectors that we evoked two weeks ago in order to avoid a full mesh? Or are we leaving this to the downstream usage of the feature? We also talked about having the AS number configurable, in a similar way that calico does.